bin/ycsb.sh

```diff
@@ -0,0 +1,183 @@
+
+# Only set YCSB_HOME if not already set
+[ -z "$YCSB_HOME" ] && YCSB_HOME=`cd "$SCRIPT_DIR/.." 2>/dev/null; pwd`
+
+# Ensure that any extra CLASSPATH variables are set via setenv.sh
+CLASSPATH=
+
+# Pull in customization options
+if [ -r "$YCSB_HOME/bin/setenv.sh" ]; then
+  . "$YCSB_HOME/bin/setenv.sh"
+fi
+
+# Attempt to find the available JAVA, if JAVA_HOME not set
+if [ -z "$JAVA_HOME" ]; then
+  JAVA_PATH=`which java 2>/dev/null`
+  if [ "x$JAVA_PATH" != "x" ]; then
+    JAVA_HOME=`dirname $(dirname "$JAVA_PATH" 2>/dev/null)`
+  fi
+fi
+
+# Determine YCSB command argument
+if [ "load" = "$1" ] ; then
+  YCSB_COMMAND=-load
+  YCSB_CLASS=com.yahoo.ycsb.Client
+elif [ "run" = "$1" ] ; then
+  YCSB_COMMAND=-t
+  YCSB_CLASS=com.yahoo.ycsb.Client
+elif [ "shell" = "$1" ] ; then
+  YCSB_COMMAND=
+  YCSB_CLASS=com.yahoo.ycsb.CommandLine
+else
+  echo "[ERROR] Found unknown command '$1'"
+  echo "[ERROR] Expected one of 'load', 'run', or 'shell'. Exiting."
+  exit 1;
+fi
+
+# Find binding information
+BINDING_LINE=`grep "^$2:" "$YCSB_HOME/bin/bindings.properties" -m 1`
+
+if [ -z $BINDING_LINE ] ; then
+  echo "[ERROR] The specified binding '$2' was not found.  Exiting."
+  exit 1;
+fi
+
+# Get binding name and class
+BINDING_NAME=`echo $BINDING_LINE | cut -d':' -f1`
+BINDING_CLASS=`echo $BINDING_LINE | cut -d':' -f2`
+
+# Some bindings have multiple versions that are managed in the same directory.
+#   They are noted with a '-' after the binding name.
+#   (e.g. cassandra-7 & cassandra-8)
+BINDING_DIR=`echo $BINDING_NAME | cut -d'-' -f1`
+
+# The 'basic' binding is core functionality
+if [ $BINDING_NAME = "basic" ] ; then
+  BINDING_DIR=core
+fi
+
+# For Cygwin, ensure paths are in UNIX format before anything is touched
+if $CYGWIN; then
+  [ -n "$JAVA_HOME" ] && JAVA_HOME=`cygpath --unix "$JAVA_HOME"`
+  [ -n "$CLASSPATH" ] && CLASSPATH=`cygpath --path --unix "$CLASSPATH"`
+fi
+
+# Check if source checkout, or release distribution
+DISTRIBUTION=true
+if [ -r "$YCSB_HOME/pom.xml" ]; then
+  DISTRIBUTION=false;
+fi
+
+# Add Top level conf to classpath
+if [ -z "$CLASSPATH" ] ; then
+  CLASSPATH="$YCSB_HOME/conf"
+else
+  CLASSPATH="$CLASSPATH:$YCSB_HOME/conf"
+fi
+
+# Build classpath
+if $DISTRIBUTION; then
+  # Core libraries
+  for f in `ls "$YCSB_HOME"/lib/*.jar`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+  # Database conf dir
+  if [ -r "$YCSB_HOME/$BINDING_DIR-binding/conf" ] ; then
+    CLASSPATH="$CLASSPATH:$YCSB_HOME/$BINDING_DIR-binding/conf"
+  fi
+
+  # Database libraries
+  for f in `ls "$YCSB_HOME"/$BINDING_DIR-binding/lib/*.jar 2>/dev/null`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+else
+  # Core libraries
+  for f in `ls "$YCSB_HOME"/core/target/*.jar`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+  # Database conf (need to find because location is not consistent)
+  for f in `find "$YCSB_HOME"/$BINDING_DIR -name "conf"`; do
+    CLASSPATH="$CLASSPATH:$f"
+  done
+
+  # Database libraries
```

Comment:
I thought about that.  I was really trying to avoid using tools that are not part of a standard linux distro.  However if you have got the source ycsb, then you'll need to have maven to build it.  How about I put in a check for the core lib and print out a message that says something to the effect of "hey, you need to build first"?

I'm hesitant to put mvn in there because it is so much faster to just grab the jars then make a call to maven each time.  As a developer, I'll call mvn when needed.  Otherwise, I just want the script to grab the available jars and run.  That's my preference.